File: 20210208/Class0208/Class0208/Car.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Class0208
{
    class Car
    {
        // 클래스 상수 변수(클래스변수화)
        public const string DAERI = "동대구 영업소";

        // 인스턴스 변수(캡슐화)
        private string company;
        private string color;
        private string model;
        private string price;

        // 기본 생성자
        public Car()
        {
              
        }

        // 일반 생성자(오버로딩)
        public Car(string company, string color, string model, string price)
        {
            // this의 역할 : 다른 변수와의 충돌 방지, 객체의 시작 위치
            this.company = company;
            this.color = color;
            this.model = model;
            this.price = price;
        } 

        public string Company
        {
            get { return company; }
            set { company = value; }
        }

        public string Color { get => color; set => color = value; }
        public string Model { get => model; set => model = value; }
        public string Price { get => price; set => price = value; }

        public void showCarInfo()
        {
            Console.WriteLine("제조회사 : " + company);
            Console.WriteLine("색상 : " + color);
            Console.WriteLine("모델명 : " + model);
            Console.WriteLine("가격 : " + price);
            Console.WriteLine("지점 : " + DAERI);
            Console.WriteLine("--------------------------------");
        }
    }
}

File: 20210112/20210113_JavaStudy/src/Car/Car.java
package Car;

public class Car {
       public static final String JIJUM = "동대구 영업소";
       private String model;
       private String company;
       private String color;
       private String price;
       
	public Car() {
		model = "K7";
		company = "기아";
		color = "블랙";
		price = "사천만원";
	}

	public Car(String model, String company, String color, String price) {
		super();
		this.model = model;
		this.company = company;
		this.color = color;
		this.price = price;
	}
	
	
	public String getModel() {
		return model;
	}

	public void setModel(String model) {
		this.model = model;
	}

	public String getPrice() {
		return price;
	}

	public void setPrice(String price) {
		this.price = price;
	}

	public void showCarInfo() {
		System.out.println("========================");
		System.out.println(Car.JIJUM);
		System.out.println("========================");
		System.out.println("모델 : " + model);
		System.out.println("회사 : " + company);
		System.out.println("색깔 : " + color);
		System.out.println("가격 : " + price);
		System.out.println("========================");
	}
       
       
}
